### DISCUSSION TITLE
v1.17.0

### DISCUSSION THREAD

**Discussion by lmakarov** [docksal/docksal#1660] [General]
# 1.17.0 (2022-04-15)

## New software versions ✨ 

- fin v1.110.1
- Docker Desktop v4.4.2+ (build 73305)
- Docker v20.10.12 (boot2docker pinned at v19.03.5)
- Docker Compose v2.1.0
- VirtualBox v6.1.32
- New multi-arch (amd64/arm64) image releases
  - [docksal/vhost-proxy v1.8](https://github.com/docksal/service-vhost-proxy/releases)
  - [docksal/ssh-agent v1.4 ](https://github.com/docksal/service-ssh-agent/releases)
  - [docksal/dns v1.2](https://github.com/docksal/service-dns/releases)
  - [docksal/cli v3.2](https://github.com/docksal/service-cli/releases)
  - [docksal/apache v2.5](https://github.com/docksal/service-apache/releases)
  - [docksal/nginx v1.2](https://github.com/docksal/service-nginx/releases)
  - [docksal/mysql v2.0](https://github.com/docksal/service-mysql/releases)
  - [docksal/mariadb v1.3 ](https://github.com/docksal/service-mariadb/releases)
  - [docksal/varnish v3.1](https://github.com/docksal/service-varnish/releases)
- PHP 8.1 is now the default version in the `default` stack and in `fin run cli` command
  - See docs on [switching/pinning PHP versions](https://docs.docksal.io/service/cli/settings/#php-versions)

## Changes and improvements ⚙️ 

- 🎉 Apple M1 and Linux ARM64 support 🎉 
  - All services in the Docksal catalog (except `solr` and `mailhog`) now use multi-arch (amd64/arm64) images
- Switched to Docker Compose v2 (v1 support dropped)
  - Running v2 in compatibility mode (forces the use underscores instead of hyphens in container names (like in v1)).
- Use of `.docksal` base domain along with `docksal-dns` internal DNS resolver is deprecated
  - Follow the [instructions](https://docs.docksal.io/stack/configuration-variables/#docksal-dns-disabled) to switch to the new public base domain - `.docksal.site`
- New `fin share-v2` command
  - Creates a temporary public URL for the project using a free Cloudflare Tunnel
- Project auto-start is now disabled by default for local environments (still ON by default for CI environments)
- Included Docksal version in `fin --version` output
  - no more confusion which Docksal vs fin version you are running
- Gitpod Integration
- `fin pull` is now an [addon](https://github.com/docksal/addons) (removed from fin core)
- Fixed VirtualBox support on macOS 12 Monterey
- Removed Drupal 8 options from `fin project create` (Drupal 8 is EOL)
- Bumped all services to the most recent image tags
  - We'll stick to the most recent versions by default and pin versions in hosting platform specific stacks
  - Current default versions:
    - Apache 2.4
    - Nginx 1.20
    - MySQL 8.0
    - MariaDB 10.6
    - Postgres 14.2
    - PHP 8.1
    - Varnish 7.0
    - Memcached 1.6
    - Redis 6.2
    - Solr 8.6-edge (legacy image, likely to be deprecated)
    - Mailhog 1.0.1
    - Elasticsearch 8.1.2
    - cloudflared `latest` (has to stay on `latest` as Cloudflare makes frequent updates and blocks older client versions)
- Switched to MariaDB 10.4 as the default `db` service in all hosting platform stacks
  - MariaDB 10.4 functions as limited drop-in replacements for MySQL 5.7, as far as InnoDB is concerned
- Updated Acquia stack versions
  - Apache 2.4
  - MariaDB 10.4 (MySQL 5.7 drop-in replacement)
  - PHP 7.4
  - Varnish 6.6
  - Memcached 1.6
  - Apache Solr 4.5
- Updated Pantheon stack image versions
  - Nginx 1.20
  - MariaDB 10.4 (MySQL 5.7 drop-in replacement)
  - PHP 7.4
  - Varnish 6.6
  - Redis 5.0
  - Apache Solr 3.6
- Updated Platform.sh stack image versions
  - Nginx 1.20
  - MariaDB 10.4 (MySQL 5.7 drop-in replacement)
  - PHP 7.4
  - Redis 5.0
- Switched to Blackfire Agent v2 in stacks (requires docksal/cli [v3.1.0](https://github.com/docksal/service-cli/releases/tag/v3.1.0)+)

## Documentation 📖 

- Expanded docs around project auto start/stop features
- Added instructions for installing an addon globally
- Added documentation on composer memory issues
- Reorganized and updated help sections in fin + misc updates in `fin help`
- Added disk space issue to common issues
- Updated MailHog docs 
- Added Github Actions build badge in README.md
- Added section on common issues for Apple MDM (Mobile Device Management) troubleshooting
- Updated docs around disabling built-in DNS resolver (docksal-dns)
- Added Cloudflare Tunnel (`fin share-v2`) docs
- Added example on how to override `solr` service image
- Added sponsor documentation

## Contributors 🤓 

- @lmakarov, @sean-e-dietrich, @shelane, @froboy, @waako, @marvoey, @paulsheldrake, @vladtulku, @JDDoesDev 

## Sponsors 🤑 

- @markaspot, @froboy, @lpeabody, @alexander-danilenko, @paulsheldrake, @610, @twfahey1, @johnoltman65
- [❤️ You ❤️ ](https://github.com/sponsors/docksal)

<hr /><em>This discussion was created from the release <a href='https://github.com/docksal/docksal/releases/tag/v1.17.0'>v1.17.0</a>.</em>

**Comment by lmakarov** [comment]
Answering [a question](https://github.com/docksal/docksal/pull/1656#issuecomment-1110672383) from @jovan1348:

> Is there maybe any way to keep using (temporary) .docksal domain in old projects?

`.docksal` base domain is deprecated, not dropped. It will continue working (and reminding about being deprecated) until the v2 is released. You have plenty of time to update old projects to use the new base domain: `.docksal.site`.

Also, it will always be possible to [manually manage DNS resolution](https://docs.docksal.io/core/system-dns/#manual) for arbitrary hosts (using `fin hosts`).

**Comment by mikeohara** [comment]
Is there a method to downgrade Docksal? I just had my machine replaced due to a hardware failure, and without thinking i just installed  Docksal to the latest version. Projects I work on, however do not work with the most recent Docksal version yet.

**Reply by shelane** [reply]
Is it just the images that your projects are not ready for? You can pin a specific version of any of the images? Do you need PHP 7.4 instead of 8.1? https://docs.docksal.io/service/cli/settings/#php-versions

**Comment by aquuleus** [comment]
Hi,

I have an issue running this update on MacOS. It seems that my dns-proxy service and all of the containers are running on **0.0.0.0** instead of **192.168.64.100**  therefore, I'm not able to reach any of them.

Is there anything I should do on my end to fix that issue ?

Edit: This occurs when: DOCKSAL_DNS_DISABLED="1"
Otherwise it works normally

Thanks

**Comment by kenrbnsn** [comment]
I tried to update Fin from  1.107.1 --> 1.110.1 on Windows 10 running WSL2 & Docker Desktop 
![image](https://user-images.githubusercontent.com/43186/168668640-2672091e-e785-47b4-9df1-99b15d4d416f.png)

This is what happened on the Ubuntu-20.04 system:
$ fin update
[STEP 1/4] Updating fin...
fin 1.110.1 downloaded...
[STEP 2/4] Updating stack files...
overrides-cloudflared.yml
overrides-gitpod.yml
overrides-dd-bind.yml
overrides-ide.yml
overrides-xhprof.yml
services.yml
stack-acquia.yml
stack-default.yml
stack-default-nodb.yml
stack-node.yml
stack-pantheon.yml
stack-platformsh.yml
volumes-bind.yml
volumes-nfs.yml
volumes-none.yml
volumes-unison.yml
Configuring network settings...
[STEP 3/4] Updating tools...
 * Starting Docker: docker                                                                                                                                                                                                                                                                                            [ OK ]
Downloading Docker...
Stopping docker service...
 * Stopping Docker: docker                                                                                                                                                                                                                                                                                            [ OK ]
# Executing docker install script, commit: 614d05e0e669a0577500d055677bb6f71e822356
Warning: the "docker" command appears to already exist on this system.

If you already have Docker installed, this script can cause trouble, which is
why we're displaying this warning and provide the opportunity to cancel the
installation.

If you installed the current Docker package using this script and are using it
again to update Docker, you can safely ignore this message.

You may press Ctrl+C now to abort this script.
+ sleep 20
^C

Then
$ fin update
$ fin restart
Disconnecting project network...
Stopping services...
Stopping socal_web_1 ... done
Stopping socal_cli_1 ... done
Starting services...
Creating network "socal_default" with the default driver
Pulling cli (docksal/cli:php8.1-3.2)...
/usr/bin/docker-credential-desktop.exe: Invalid argument
Traceback (most recent call last):
  File "docker/credentials/store.py", line 80, in _execute
  File "subprocess.py", line 411, in check_output
  File "subprocess.py", line 512, in run
subprocess.CalledProcessError: Command '['/usr/bin/docker-credential-desktop.exe', 'get']' returned non-zero exit status 1.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "docker/auth.py", line 264, in _resolve_authconfig_credstore
  File "docker/credentials/store.py", line 35, in get
  File "docker/credentials/store.py", line 93, in _execute
docker.credentials.errors.StoreError: Credentials store docker-credential-desktop.exe exited with "".

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "docker-compose", line 3, in <module>
  File "compose/cli/main.py", line 81, in main
  File "compose/cli/main.py", line 203, in perform_command
  File "compose/metrics/decorator.py", line 18, in wrapper
  File "compose/cli/main.py", line 1186, in up
  File "compose/cli/main.py", line 1182, in up
  File "compose/project.py", line 664, in up
  File "compose/service.py", line 358, in ensure_image_exists
  File "compose/service.py", line 1251, in pull
  File "compose/progress_stream.py", line 99, in get_digest_from_pull
  File "compose/service.py", line 1218, in _do_pull
  File "docker/api/image.py", line 411, in pull
  File "docker/auth.py", line 48, in get_config_header
  File "docker/auth.py", line 324, in resolve_authconfig
  File "docker/auth.py", line 235, in resolve_authconfig
  File "docker/auth.py", line 281, in _resolve_authconfig_credstore
docker.errors.DockerException: Credentials store error: StoreError('Credentials store docker-credential-desktop.exe exited with "".')
[2215] Failed to execute script docker-compose

Now I can't run any of my projects. :(

Ken
